docker/Dockerfile

```diff
@@ -23,38 +23,44 @@ RUN apt-get -y install  xvfb xserver-xephyr libxml2-dev libxslt-dev libssl-dev z
   python-pip postgresql-server-dev-all postgresql-client postgresql-client-common \
   postgresql libcurl4-openssl-dev proxychains tor ca-certificates libpq-dev python-dev \
   libxslt1-dev libldap2-dev libsasl2-dev libffi-dev net-tools lsof locales-all \
-  libyaml-dev python3 python3-pip python3.8-dev build-essential
-
-RUN update-alternatives --install /usr/bin/python python /usr/bin/python3 1
+  libyaml-dev python3 python3-pip python3.7-dev build-essential python3-virtualenv python3-lxml
 
 # Set locale
 ENV LANG en_US.UTF-8
 ENV LANGUAGE en_US:en
 ENV LC_ALL en_US.UTF-8
 
 # Install optional tools (LBD, arachni, gnutls-bin, o-saft and metagoofil)
-RUN apt-get -y install  lbd arachni theharvester tlssled nikto dnsrecon nmap whatweb skipfish dirbuster \
+RUN apt-get -y install lbd theharvester tlssled nikto dnsrecon nmap whatweb skipfish dirbuster \
   metasploit-framework wpscan wapiti hydra metagoofil o-saft
 
 # Ensure pip and setuptools are at their latest versions
-RUN pip install setuptools --upgrade
-RUN pip install cffi
+RUN pip3 install setuptools --upgrade
+RUN pip3 install cffi
+RUN python3 -m pip install --upgrade pip
 
-# Install and create a Python virtualenv
-RUN pip install virtualenv && virtualenv ~/.venv/owtf && . ~/.venv/owtf/bin/activate
+#Create and activating a python virtualenv
+ENV VIRTUAL_ENV=/opt/owtf
+RUN python3 -m virtualenv --python=/usr/bin/python3.7 $VIRTUAL_ENV
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
 # Create a dedicated OWTF directory to copy source to and run from.
 RUN cd / && /bin/mkdir -p owtf
 ADD . /owtf
+ADD docker/wait-for-it.sh /usr/bin/wait-for-it.sh
+RUN chmod +x /usr/bin/wait-for-it.sh
 
-RUN pip install -r /owtf/requirements/base.txt
+#Installing python dependencies
+RUN pip3 install -r /owtf/requirements/base.txt
 
 # Install OWTF using the recommended method (setup.py)
 RUN cd /owtf &&\
-  python setup.py develop
+  python setup.py install
```

Comment:
you should keep this to `develop` since this is a dev Docker container - any changes you do while developing will automatically reflect otherwise you have to install the package every time to see your changes.

Reply:
I have made the changes. Thank You.